### PR TITLE
Word2vec running on current Keras

### PR DESCRIPTION
**Pull request by SimonPavlik**
Word2veckeras word2vec running on current version of Keras using its Functional API model.

Things to do:
- [ ] convert doc2vec to current Keras
- [ ] test with Tensorflow back-end

### CODE REVIEW

**Comment by wddabc** [comment]
Hi Simon,
Great work. This is exactly what am I looking for (I'm building something that needs a keras version of skip-gram). I have few cents worth. 

1. There is a typo in https://github.com/SimonPavlik/word2vec-keras-in-gensim/blob/8cccc4b85be1655285a8d11096c34b90941a1ced/word2veckeras/word2veckeras.py#L105, where it should be 'mul' rather than 'sum'.

2. Any reason you use the 'mse' loss rather than the 'binary_crossentropy' I think the later one is what from gensim as well as Mikolov's implementation.

3. Negative sampling can be done in O(1) by [alias sampling](https://pandasthumb.org/archives/2012/08/lab-notes-the-a.html).

4. I saw your [blog](https://rare-technologies.com/gensim-word2vec-on-cpu-faster-than-word2veckeras-on-gpu-incubator-student-blog/) about the efficiency issue of Word2Veckeras on Theano. I changed 
`self.kerasmodel.fit_generator(gen,samples_per_epoch=samples_per_epoch, nb_epoch=self.iter,verbose=0)`
to 
` for epoch in xrange(self.iter):
       print 'epoch %i, %f' % (epoch, self.kerasmodel.train_on_batch(*next(gen)))`
and got a lot of speedups.

**Comment by SimonPavlik** [comment]
@wddabc Hi Dingquan:

Thank you for reviewing the code and the tips! Did you come close to Gensim's W2V speed with Your speedup?

**Comment by pkmandke** [comment]
@SimonPavlik 
I guess this is a super old thread but did you test with a Tensorflow backend for this repo?

**Comment by SimonPavlik** [comment]
@pkmandke I think I did, but it's been quite some time. I also ran some benchmarks then with results on [RaRe Technologies blog](https://rare-technologies.com/gensim-word2vec-on-cpu-faster-than-word2veckeras-on-gpu-incubator-student-blog/).
